File: packages/python/src/massed_compute_mcp/commands/config_show.py
"""`massed-compute-mcp config show` — inspect stored config."""

from __future__ import annotations

from ..config import DEFAULT_BASE_URL, config_path, read_config, resolve_auth


def _mask(key: str) -> str:
    if len(key) <= 8:
        return "***"
    return f"{key[:4]}…{key[-4:]}"


def run_config_show(_argv: list[str]) -> int:
    file = config_path()
    print(f"Config file: {file}")
    exists = file.exists()
    print(f"Exists:      {'yes' if exists else 'no'}")

    if exists:
        stored = read_config()
        if stored:
            print(f"Stored key:  {_mask(stored.get('apiKey', ''))}")
            print(f"Base URL:    {stored.get('baseUrl') or DEFAULT_BASE_URL}")
            if stored.get("validatedAt"):
                print(f"Validated:   {stored['validatedAt']}")
        else:
            print("Stored key:  (file exists but is malformed; run `init` to overwrite)")
    print()

    auth = resolve_auth()
    print("Active resolution (what the MCP server would use right now):")
    print(f"  source:    {auth.source}")
    if auth.api_key:
        print(f"  key:       {_mask(auth.api_key)}")
    print(f"  base url:  {auth.base_url}")
    if auth.source == "none":
        print()
        print("Run `massed-compute-mcp init` to set a key.")
    return 0

File: packages/node/src/commands/config-show.ts
/**
 * `massed-compute-mcp config show`
 *
 * Print where the config lives and a masked view of what's stored. Useful
 * for "where is my config again?" questions and for confirming the key
 * resolution chain without making a live upstream call (that's what
 * `doctor` is for).
 */

import * as fs from "node:fs";
import { configPath, readConfig, resolveAuth, DEFAULT_BASE_URL } from "../config.js";

const mask = (key: string): string => {
  if (key.length <= 8) return "***";
  return `${key.slice(0, 4)}…${key.slice(-4)}`;
};

export const runConfigShow = async (_argv: string[]): Promise<number> => {
  const file = configPath();
  console.log(`Config file: ${file}`);
  const exists = fs.existsSync(file);
  console.log(`Exists:      ${exists ? "yes" : "no"}`);

  if (exists) {
    const stored = readConfig();
    if (stored) {
      console.log(`Stored key:  ${mask(stored.apiKey)}`);
      console.log(`Base URL:    ${stored.baseUrl ?? DEFAULT_BASE_URL}`);
      if (stored.validatedAt) console.log(`Validated:   ${stored.validatedAt}`);
    } else {
      console.log("Stored key:  (file exists but is malformed; run `init` to overwrite)");
    }
  }
  console.log("");

  const auth = resolveAuth();
  console.log("Active resolution (what the MCP server would use right now):");
  console.log(`  source:    ${auth.source}`);
  if (auth.apiKey) console.log(`  key:       ${mask(auth.apiKey)}`);
  console.log(`  base url:  ${auth.baseUrl}`);
  if (auth.source === "none") {
    console.log("");
    console.log("Run `massed-compute-mcp init` to set a key.");
  }
  return 0;
};
